backend/lib/richard_burton/publication.ex

```diff
@@ -65,10 +64,57 @@ defmodule RichardBurton.Publication do
   end
 
   def insert(attrs) do
-    %__MODULE__{} |> changeset(attrs) |> Repo.insert()
+    %__MODULE__{}
+    |> changeset(attrs)
+    |> Repo.insert()
+    |> case do
+      {:ok, publication} ->
+        {:ok, publication}
+
+      {:error, changeset} ->
+        {:error, get_errors(changeset)}
+    end
+  end
+
+  def validate(attrs) do
+    changeset = changeset(%__MODULE__{}, attrs)
+
+    if changeset.valid? do
+      unique_key = [:title, :country, :year, :publisher]
+      unique_key_values = Repo.get_unique_key_values(unique_key, changeset)
+
+      __MODULE__
+      |> Repo.get_by(Enum.zip(unique_key, unique_key_values))
+      |> case do
+        nil -> {:ok}
```

Comment:
It's more idiomatic to return `:ok` instead of `{:ok}`

Reply:
This was a long time ago, but I guess it was meant to be `{:ok, attrs}`, the test descriptions says that 🤔

Reply:
It's sad because this function ended up unused in this PR 😢 sorry for that

Reply:
I'll do {:ok, attrs} for now because it makes sense as an interface if you want to keep processing in the same pipe after validation